nixos/modules/services/monitoring/opentelemetry-collector.nix

```diff
@@ -60,29 +60,29 @@ in
     systemd.services.opentelemetry-collector = {
       description = "Opentelemetry Collector Service Daemon";
       wantedBy = [ "multi-user.target" ];
-
-      serviceConfig =
+      script =
         let
           conf =
             if cfg.configFile == null then
               settingsFormat.generate "config.yaml" cfg.settings
             else
               cfg.configFile;
         in
-        {
-          ExecStart = "${getExe opentelemetry-collector} --config=file:${conf}";
-          DynamicUser = true;
-          Restart = "always";
-          ProtectSystem = "full";
-          DevicePolicy = "closed";
-          NoNewPrivileges = true;
-          WorkingDirectory = "%S/opentelemetry-collector";
-          StateDirectory = "opentelemetry-collector";
-          SupplementaryGroups = [
-            # allow to read the systemd journal for opentelemetry-collector
-            "systemd-journal"
-          ];
-        };
+        "${getExe opentelemetry-collector} --config=file:${conf} $1";
```

Comment:
I don't see how this would work. The command ends up being written into a bash wrapper that gets written to `ExecStart` instead of the command being placed there directly. If you could append arguments behind that script in `ExecStart` via some flag in `systemctl`, it would be the same as if you appended those arguments directly to the command, no? Or am I missing something?